gcp/service_auth/tests/test_get_token.py

```diff
@@ -7,33 +8,47 @@
 import logging
 import datetime
 from pathlib import Path
+import json
 
-from gcp_service_auth import GcpIdentityToken
+try:
+    from gcp_service_auth import GcpIdentityToken
+except ModuleNotFoundError:
+    sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+    from gcp_service_auth import GcpIdentityToken
+    pass
+                                 
 
-TEST_URL = os.environ.get('TEST_URL')
 
 @pytest.fixture(scope="module")
 #@pytest.mark.with_op
-def gcp_browse_cred() -> Generator[str, str, str]:
+def gcp_browse_cred() -> (Generator[str, str, str], str):
     logging.basicConfig(level=logging.DEBUG)
     cred_file = os.path.join(Path(__file__).parent, "browse-local.json")
     cred_file = cred_file
     if not os.path.exists(cred_file):
         subprocess.run(["op", "document", "get", "4feibaz4tzn6iwk5c7ggvb7xwi", "-o", cred_file])
     os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = cred_file
-    yield cred_file
+    op = subprocess.run(["op", "item", "get", "4feibaz4tzn6iwk5c7ggvb7xwi", "--format", "json"], stdout=subprocess.PIPE)
+    test_meta = json.loads(op.stdout)
+    test_url = ""
+    for field in test_meta['fields']:
+        if field["id"] == "heltf7phky3h6rnlvd7zi3542u":
+            test_url = field["value"]
+            break
+    yield (cred_file, test_url)
     os.remove(cred_file)
     return ""
```

Comment:
Not your code, but IMHO "file" and "filename" are well-understood, distinct data types.  I would call this "cred_filename" not "cred_file".  

What is the purpose of assigning cred_file = cred_file ?

Reply:
What did the magic strings 4feibaz4tzn6iwk5c7ggvb7xwi and heltf7phky3h6rnlvd7zi3542u come from?

Reply:
It is the output from 1password CLI "op". The cred in 1p has these keys.